src/components/fields/Percentage/Settings.tsx

```diff
@@ -0,0 +1,195 @@
+import { useState } from "react";
+
+import {
+  Box,
+  ButtonBase,
+  Checkbox,
+  Collapse,
+  InputLabel,
+  useTheme,
+} from "@mui/material";
+import ColorPickerInput from "@src/components/ColorPickerInput";
+import { ISettingsProps } from "@src/components/fields/types";
+
+import { Color, toColor } from "react-color-palette";
+import { fieldSx } from "@src/components/SideDrawer/utils";
+import { ChevronDown } from "mdi-material-ui";
+import { ReactElement } from "react-markdown/lib/react-markdown";
+import { resultColorsScale } from "@src/utils/color";
+
+const ColorPickerCollapse = ({
+  colorKey,
+  color,
+  active,
+  setActive,
+  disabled,
+  children,
+}: {
+  colorKey: string;
+  color: Color;
+  active: boolean;
+  setActive: (activeKey: string | null) => void;
+  disabled?: boolean;
+  children?: ReactElement;
+}) => {
+  const toggleCollapse = () => !disabled && setActive(active ? null : colorKey);
```

Comment:
Better to lift this logic to settings, so there'll be no need to pass active, setActive, or color key. Send only an onClickHandler instead.